### PR TITLE
Find cell method

### PR DESCRIPTION
**Pull request by jRoeltgen**
Find cell binary search method in rect_grid.

### CODE REVIEW

File: zero/gkyl_rect_grid.h [hunk 1]
```diff
@@ -29,6 +29,20 @@ struct gkyl_rect_grid {
 void gkyl_rect_grid_init(struct gkyl_rect_grid *grid, int ndim,
   const double *lower, const double *upper, const int *cells);
 
+/**
+ * Find cell indices of point
+ *
+ * @param grid Grid object
+ * @param point The point to find the cell indices at
+ * @param pick_lower
```
Comment (manauref) [inline zero/gkyl_rect_grid.h:37]: @jRoeltgen can you add a description for this argument please?

File: zero/gkyl_rect_grid_priv.h [hunk 2]
```diff
@@ -0,0 +1,55 @@
+#pragma once
+
+#include <gkyl_rect_grid.h>
+
+/* Checks if given point is in given cell
```
Comment (manauref) [inline zero/gkyl_rect_grid_priv.h:5]: @jRoeltgen please revise these comments, I think they are not right for in_dir

File: zero/gkyl_rect_grid.h (hunk 1 above)
Comment (jRoeltgen) [reply zero/gkyl_rect_grid.h]: Added "If point on cell boundary, pick lower cell if true, and upper if false"

File: zero/gkyl_rect_grid_priv.h (hunk 2 above)
Comment (jRoeltgen) [reply zero/gkyl_rect_grid_priv.h]: Changed "Checks if given point is in given cell" to "Find upper and lower boundaries of given cell"

**Pull request merged**